.pre-commit-config.yaml

```diff
@@ -3,57 +3,58 @@ default_language_version:
 
 repos:
 -   repo: https://github.com/asottile/pyupgrade
-    rev: v2.11.0
+    rev: v3.3.1
     hooks:
     -   id: pyupgrade
         language_version: python3
-# Irrelevant hook for the time being
-# -   repo: https://github.com/kynan/nbstripout
-#     rev: 0.3.9
-#     hooks:
-#     -   id: nbstripout
-#         language_version: python3
-#         files: ".ipynb"
-#         args: ["--keep-output"]
+
 -   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v3.4.0
+    rev: v4.4.0
     hooks:
-    -   id: trailing-whitespace
-        language_version: python3
-        exclude: .txt|.sld
-    -   id: end-of-file-fixer
-        language_version: python3
-        exclude: .ipynb|.txt|.sld
     -   id: check-yaml
         language_version: python3
         exclude: conda-recipe/meta.yaml
     -   id: debug-statements
         language_version: python3
+    -   id: end-of-file-fixer
+        language_version: python3
+        exclude: .ipynb|.txt|.sld
+    -   id: trailing-whitespace
+        language_version: python3
+        exclude: .txt|.sld
+
 -   repo: https://github.com/ambv/black
-    rev: 20.8b1
+    rev: 22.12.0
     hooks:
     -   id: black
-        language_version: python3
-        args: ["--target-version", "py36"]
+
```

Comment:
`geoserver-rest` is backwards compatible to Python3.6. There are some breaking changes in Python3.7 (trailing commas in function call signatures) that require us to specify `py36` otherwise there will be issues.

Reply:
Python 3.6 is EOL for 1 year and 3.7 will be EOL too at next summer https://endoflife.date/python.

Are you sure?

Reply:
Good point. In that case, we should explicitly drop support for Python3.6 in the setup.py.

@iamtekson would you be alright with making that change here?

Reply:
> I revert the removal of line 37 

Done.